CDC.WebNotes.Api/Controllers/AuthController.cs

```diff
@@ -0,0 +1,72 @@
+﻿using CDC.WebNotes.Application.Contracts;
+using CDC.WebNotes.Application.Services;
+using CDC.WebNotes.Data;
+using CDC.WebNotes.Domain.Notes;
+using CDC.WebNotes.Dto;
+using CDC.WebNotes.Dto.Notes;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WebNotes.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthService _authService;
+
+        public AuthController(IAuthService authService)
+        {
+            _authService = authService;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("login")]
+        public async Task<IActionResult> Login(string username, string password, [FromServices] IHttpContextAccessor httpContextAccessor)
+        {
+            var user = await _authService.GetUserByUsernameAsync(username);
+            if (user == null || !_authService.VerifyPasswordHash(password, user.PasswordHash))
+            {
+                return Unauthorized();
+            }
+
+            var claimIdentity = new ClaimsIdentity(new Claim[] {
+                new Claim(ClaimTypes.Name, user.Username),
+                new Claim(ClaimTypes.Role, "User")
+               }, CookieAuthenticationDefaults.AuthenticationScheme);
+
+            var httpContext = httpContextAccessor.HttpContext;
+
+            await httpContext.SignInAsync(CookieAuthenticationDefaults.AuthenticationScheme, new ClaimsPrincipal(claimIdentity));
+
+            return Ok();
+        }
+ 
+        [AllowAnonymous]
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(string username, string password)
+        {
+            var user = await _authService.GetUserByUsernameAsync(username);
+            if (user != null)
+            { 
+                return BadRequest("User with that name already exists");
```

Comment:
```suggestion
                return BadRequest($"User with name {user.Username} already exists");
```
Please try all the times to clarify data and values that is not valid.